File: kotlin/gradle-conventions/src/main/kotlin/software/onepiece/gradle/pluginshowto/MyJavaLibraryPlugin.kt
package software.onepiece.gradle.pluginshowto

import org.gradle.api.Plugin
import org.gradle.api.Project
import org.gradle.api.plugins.JavaLibraryPlugin

abstract class MyJavaLibraryPlugin : Plugin<Project> {

    override fun apply(project: Project): Unit = with(project) {
        plugins.apply(MyJavaBasePlugin::class.java)
        plugins.apply(JavaLibraryPlugin::class.java)
    }
}

File: clojure/gradle-conventions/src/main/clojure/software/onepiece/gradle/pluginshowto/MyJavaLibraryPlugin.clj
(ns software.onepiece.gradle.pluginshowto.MyJavaLibraryPlugin
    (:import
      (org.gradle.api Project Plugin)
      (org.gradle.api.plugins JavaLibraryPlugin)
    )
    (:gen-class
      :name       software.onepiece.gradle.pluginshowto.MyJavaLibraryPlugin
      :implements [org.gradle.api.Plugin]
    )
  )

(defn -apply [this ^Project project]
  (do
    (.apply (.getPlugins project) "software.onepiece.gradle.pluginshowto.java-base")
    (.apply (.getPlugins project) JavaLibraryPlugin)
  )
)

